src/com/shashi/service/impl/ProductServiceImpl.java

```diff
@@ -633,6 +727,41 @@ public String updateProductWithoutImage(String prevProductId, ProductBean update
 		return status;
 	}
 
+
+	// Method to check if a discount already exists for a product
+	private boolean discountExists(String productId) {
```

Comment:
no need to call the db. you can just check if product.getDiscount != null. also if you use the query from the other comment you should be able to completely delete this method